plugin/pkg/shoot/validator/admission.go

```diff
@@ -2135,3 +2136,38 @@ func (c *validationContext) validateManagedServiceAccountIssuer(
 
 	return nil
 }
+
+func (c *validationContext) validateLimits(a admission.Attributes) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if a.GetOperation() == admission.Delete || c.cloudProfileSpec.Limits == nil {
```

Comment:
We should probably also skip the validation in general if shoot has a deletion timestamp.

Reply:
Good point, done. PTAL :)